### PR TITLE
build: adjust the build to wire up the library search path

### PR DESCRIPTION
**Pull request by compnerd**
This is needed to support Swift macros on Windows.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: lldb/cmake/modules/AddLLDB.cmake [hunk 1]
```diff
@@ -190,17 +190,22 @@ function(add_properties_for_swift_modules target reldir)
...
       string(REGEX MATCH "^[^-]*" arch ${LLVM_TARGET_TRIPLE})
       target_link_libraries(${target} PRIVATE swiftCore-linux-${arch})
       string(TOLOWER ${CMAKE_SYSTEM_NAME} platform)
       set(SWIFT_BUILD_RPATH "${LLDB_SWIFT_LIBS}/${platform}")
       set(SWIFT_INSTALL_RPATH "$ORIGIN/${reldir}lib/swift/${platform}")
+      set_property(TARGET ${target} APPEND PROPERTY BUILD_RPATH "${SWIFT_BUILD_RPATH}")
+      set_property(TARGET ${target} APPEND PROPERTY INSTALL_RPATH "${SWIFT_INSTALL_RPATH}")
+    elseif(CMAKE_SYSTEM_NAME MATCHES Windows)
+      target_link_directories(${target} PRIVATE
+        ${SWIFT_PATH_TO_SWIFT_SDK}/usr/lib/swift/Windows/x86_64)
```
Comment (bnbarham) [inline lldb/cmake/modules/AddLLDB.cmake:206]: Does swiftCore end up being linked from somewhere? I would have expected something similar to the Linux block here (minus the RPATHs)

Comment (compnerd) [reply lldb/cmake/modules/AddLLDB.cmake:206]: `swiftCore.lib` is found in this location.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

File: lldb/cmake/modules/AddLLDB.cmake (hunk 1 above)
Comment (bnbarham) [reply lldb/cmake/modules/AddLLDB.cmake:206]: Sure, I'm just confused where that linking is actually happening 😅

**Review by bnbarham — COMMENTED** [review]
LGTM other than the hardcoded arch. Can we do the same as Linux there? Could just move the regex match up to the top of the if.

**Comment by compnerd** [comment]
I'll unhardcode the arch in a follow up.

**Comment by bulbazord** [comment]
@compnerd Can you also cherry-pick this to `stable/20230725` and `next` so we don't lose track of this change?

**Comment by compnerd** [comment]
https://github.com/apple/llvm-project/pull/7470

**Pull request merged**